src/main/java/com/yunhalee/walkerholic/activity/service/ActivityService.java

```diff
@@ -1,98 +1,88 @@
-                .get();
-            existingActivity.setName(activityCreateDTO.getName());
-            existingActivity.setDescription(activityCreateDTO.getDescription());
-            existingActivity.setScore(activityCreateDTO.getScore());
-
-            if (multipartFile != null) {
-                saveActivityImage(existingActivity, multipartFile, false);
-            }
-            activityRepository.save(existingActivity);
-            return new ActivityCreateDTO(existingActivity);
-        } else {
-            Activity activity = new Activity();
-            activity.setName(activityCreateDTO.getName());
-            activity.setDescription(activityCreateDTO.getDescription());
-            activity.setScore(activityCreateDTO.getScore());
-            activityRepository.save(activity);
-
-            if (multipartFile != null) {
-                saveActivityImage(activity, multipartFile, true);
-            }
-            activityRepository.save(activity);
-            return new ActivityCreateDTO(activity);
-        }
-
+    @Transactional
+    public ActivityResponse create(ActivityRequest activityRequest,
+        MultipartFile multipartFile) throws IOException {
+        Activity activity = activityRequest.toActivity();
+        activityRepository.save(activity);
+        saveImage(activity, multipartFile);
 
+        return new ActivityResponse(activity);
     }
 
+    @Transactional
+    public ActivityResponse update(Integer id, ActivityRequest activityRequest,
+        MultipartFile multipartFile) throws IOException {
+        Activity existingActivity = activityRepository.findById(id)
+            .orElseThrow(() -> new ActivityNotFoundException(
+                "Activity not found with id : " + id));
+        Activity requestActivity = activityRequest.toActivity();
+        Activity updatedActivity = existingActivity.update(requestActivity);
+        saveImage(updatedActivity, multipartFile);
+
+        return new ActivityResponse(existingActivity);
+    }
 
 
-    public ActivityDTO getActivity(Integer id) {
+    public ActivityDetailResponse activity(Integer id) {
         Activity activity = activityRepository.findByActivityId(id);
-        ActivityDTO activityDTO = new ActivityDTO(activity);
+        ActivityDetailResponse activityDetailResponse = new ActivityDetailResponse(activity);
 
-        return activityDTO;
+        return activityDetailResponse;
     }
 
-    public List<ActivityCreateDTO> getActivities() {
-        List<Activity> activities = activityRepository.findAll();
-        List<ActivityCreateDTO> activityCreateDTOS = new ArrayList<>();
-        activities.forEach(activity -> activityCreateDTOS.add(new ActivityCreateDTO(activity)));
-        return activityCreateDTOS;
+    public List<ActivityResponse> activities() {
+        return activityRepository.findAll().stream().map(ActivityResponse::new)
+            .collect(Collectors.toList());
```

Comment:
```suggestion
    public List<ActivityResponse> activities() {
        return activityRepository.findAll().stream()
            .map(ActivityResponse::new)
            .collect(Collectors.toList());
```
의미있는 단위로 개행하는 것은 어떨까요

Reply:
해당부분 확인하여 수정하였습니다!